plutus-core/untyped-plutus-core/testlib/Evaluation/Builtins/Definition.hs

```diff
@@ -1277,6 +1278,47 @@ test_Case =
                     Left _                        -> False
                     Right EvaluationFailure       -> 0 > scrut || scrut >= fromIntegral (length is)
                     Right (EvaluationSuccess res) -> res == mkConstant () (is !! fromIntegral scrut)
+        , QC.testProperty "List success" . QC.withMaxSuccess 99 $
+            \(scrut :: [Integer]) ->
+                let
+                  term :: Term TyName Name DefaultUni DefaultFun ()
+                  term = runQuote $ do
+                    x <- freshName "x"
+                    xs <- freshName "xs"
+                    let
+                      listElem = mkTyBuiltin @_ @Integer ()
+                      list = mkTyBuiltin @_ @[Integer] ()
+                    pure $
+                      kase ()
+                        listElem
+                        (mkConstant () scrut)
+                        [ mkConstant @Integer () 42
+                        , lamAbs () x listElem $ lamAbs () xs list $ var () x
```

Comment:
We should rename `Bool success` to `Bool success 2` and add `Bool success 1` for when there's only one branch. And do the same here. Because `Bool any` doesn't properly cover the case with a single branch and the `False` scrutinee.

But that can be a separate PR.